core/src/main/java/org/jdbi/v3/MappingRegistry.java

```diff
@@ -26,10 +27,10 @@
     private static final PrimitivesColumnMapperFactory BUILT_INS = new PrimitivesColumnMapperFactory();
 
     private final List<ResultSetMapperFactory> rowFactories = new CopyOnWriteArrayList<>();
-    private final ConcurrentHashMap<Class<?>, ResultSetMapper<?>> rowCache = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Type, ResultSetMapper<?>> rowCache = new ConcurrentHashMap<>();
```

Comment:
I worry a little bit about a `Map` with an key that is an interface that doesn't have a specified equality.  Maybe I worry too much...